File: app-core/src/main/java/ru/micron/security/UserDetailsService.java
package ru.micron.security;

import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;
import ru.micron.mapper.JwtUserMapper;
import ru.micron.service.UserService;

@Service
@RequiredArgsConstructor
public class UserDetailsService implements
    org.springframework.security.core.userdetails.UserDetailsService {

  private final UserService userService;
  private final JwtUserMapper jwtUserMapper;

  @Override
  public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
    return jwtUserMapper.toUserDetails(userService.findByUsername(username));
  }
}

File: app-core/src/main/kotlin/ru/micron/security/UserDetailsService.kt
package ru.micron.security

import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service
import ru.micron.mapper.JwtUserMapper
import ru.micron.service.UserService

@Service
class UserDetailsService(
    private val userService: UserService,
    private val jwtUserMapper: JwtUserMapper
) : UserDetailsService {

    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String): UserDetails {
        return jwtUserMapper.toUserDetails(userService.findByUsername(username))
    }
}